docs/tutorials/cooperative_groups_tutorial.rst

```diff
@@ -0,0 +1,312 @@
+.. meta::
+  :description: The tutorial of cooperative groups in HIP
+  :keywords: AMD, ROCm, HIP, cooperative groups, tutorial
+
+*******************************************************************************
+Tutorial: Cooperative Groups
+*******************************************************************************
+
+This tutorial will show you the basic concepts of the cooperative groups in HIP
+programming model, the most essential tooling around it and briefly rehash some
+commonalities of heterogenous APIs in general. Mild familiarity with the C/C++
+compilation model and the language is assumed throughout this article.
+
+Prerequisites
+=============
+
+In order to follow this tutorial you will need properly installed drivers and a
+HIP compiler toolchain to compile your code. Because HIP provided by ROCm
+supports compiling and running on Linux and Windows with AMD and NVIDIA GPUs
+alike, the combination of install instructions are more then worth covering as
+part of this tutorial. Please refer to :doc:`/install/install` on how to
+install HIP development packages.
```

Comment:
```suggestion
To follow this tutorial, you'll need properly installed drivers and a
HIP compiler toolchain to compile your code. Because ROCm HIP supports compiling and running on Linux and Microsoft Windows with AMD and NVIDIA GPUs, review the HIP development package installation before starting this tutorial. For more information, see :doc:`/install/install`.
```